universal-application-tool-0.0.1/app/forms/MultiOptionQuestionForm.java

```diff
@@ -33,16 +42,27 @@ protected MultiOptionQuestionForm(MultiOptionQuestionDefinition qd) {
     this.maxChoicesAllowed = qd.getMultiOptionValidationPredicates().maxChoicesAllowed();
 
     this.options = new ArrayList<>();
+    this.newOptions = new ArrayList<>();
+    this.optionIndexes = new ArrayList<>();
 
     try {
       // The first time a question is created, we only create for the default locale. The admin can
       // localize the options later.
       if (qd.getSupportedLocales().contains(LocalizedStrings.DEFAULT_LOCALE)) {
-        List<String> optionStrings =
-            qd.getOptionsForLocale(LocalizedStrings.DEFAULT_LOCALE).stream()
-                .map(LocalizedQuestionOption::optionText)
-                .collect(Collectors.toList());
-        this.options.addAll(optionStrings);
+        qd.getOptionsForLocale(LocalizedStrings.DEFAULT_LOCALE).stream()
+            .sorted(Comparator.comparing(LocalizedQuestionOption::order))
+            .forEachOrdered(
+                option -> {
+                  options.add(option.optionText());
+                  optionIndexes.add(option.id());
```

Comment:
(blocking) it is weird that `id`s are being set to the list of indexes, and not `option.order()`.

Reply:
Renamed to `optionIds` everywhere.